### PR TITLE
add pekko-connectors-bom

### PR DESCRIPTION
**Pull request by pjfanning**
We have BOMs for Pekko(Core) and Pekko-HTTP.

* https://repo1.maven.org/maven2/org/apache/pekko/pekko-bom_2.13/1.0.2/pekko-bom_2.13-1.0.2.pom
* https://repo1.maven.org/maven2/org/apache/pekko/pekko-http-bom_2.13/1.0.1/pekko-http-bom_2.13-1.0.1.pom

I think it useful to have one for Pekko-Connectors.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -434,6 +436,16 @@ lazy val `doc-examples` = project
     publish / skip := true,
     Dependencies.`Doc-examples`)
 
+lazy val billOfMaterials = Project("bill-of-materials", file("bill-of-materials"))
+  .enablePlugins(BillOfMaterialsPlugin)
+  .disablePlugins(MimaPlugin, SitePlugin)
+  .settings(
+    name := "pekko-connectors-bom",
+    licenses := List(License.Apache2),
+    libraryDependencies := Seq.empty,
```
Comment (Roiocam) [inline build.sbt:445]: `libraryDependencies` default value is Seq.empty? if true we could ignore this.

**Review by Roiocam — APPROVED** [review]
Looks good, Thanks. 

If we have a BOM that records all modules(more tha this), it will be better, but it will come with higher maintenance costs.

File: build.sbt (hunk 1 above)
Comment (pjfanning) [reply build.sbt:445]: this is needed to fight against the commonSettings in the build - that hardcodes a dependency on pekko-stream.
BOMs don't have dependencies.

**Comment by Roiocam** [comment]
>  My project is easier to maintain if the main pekko repos have BOMs that I can more easily aggregate.

It sounds reasonable.  Let's move forward.

**Pull request merged**